packages/block-editor/src/components/block-navigation/block-select-button.js

```diff
@@ -43,12 +40,6 @@ function BlockNavigationBlockSelectButton(
 	const blockInformation = useBlockDisplayInformation( clientId );
 	const instanceId = useInstanceId( BlockNavigationBlockSelectButton );
 	const descriptionId = `block-navigation-block-select-button__${ instanceId }`;
-	const blockType = getBlockType( name );
-	const blockLabel = getBlockLabel( blockType, attributes );
-	// If label is defined we prioritize it over possible possible
-	// block variation match title.
-	const blockDisplayName =
```

Comment:
I don't recall why we were not using `BlockTitle` here instead of calculating again cc @youknowriad

Reply:
There was a lengthy PR with a long discussion about the label here. I think @talldan was involved in it maybe? Though I don't have a direct answer.

Reply:
Looks like `BlockTitle` didn't the label when this was initially implemented, but was later updated to do so.

This is a nice improvement.